src/abstractdataframe/iteration.jl

```diff
@@ -27,25 +27,32 @@ with each row represented as a `DataFrameRow`.
 """
 eachrow(df::AbstractDataFrame) = DFRowIterator(df)
 
+Base.size(itr::DFRowIterator) = (size(itr.df, 1), )
+Base.size(itr::DFRowIterator, ix) =
```

Comment:
This method shouldn't be needed if you declare that `DFRowIterator <: AbstractVector{DataFrameRow}` according to https://docs.julialang.org/en/latest/manual/interfaces/#man-interface-array-1. Then you won't need `firstindex`, `lastindex`, `eltype` and `iterate` either.